packages/remix-server-runtime/sessions.ts

```diff
@@ -221,19 +224,24 @@ export interface SessionIdStorageStrategy<
    * Creates a new record with the given data and returns the session id.
    */
   createData: (
+    this: void,
```

Comment:
Is there a real need to `this: void` for an arrow function? I'm surprised it's even allowed - maybe another lint rule request for typescript-eslint...

Reply:
There is a difference between arrow functions with and without the `this: void`. This will affect the conversion of all the functions to arrow functions too.

With the `this: void`, TypeScript enforces the contract on the implementing function that there is no viable `this`. For example, try adding a `this` reference to an implementer, such as in `workerKVStorage.ts`:

```ts
...
    },
    async readData(id) {
      this.deleteData(id); // error when arrow functions in `SessionIdStorageStrategy` has `this: void` no error with just arrow function.
      let session = await kv.get(id);

      if (!session) {
        return null;
      }

      return JSON.parse(session);
    },
...
```

So while changing the `SessionStorage` methods to arrow functions will fix the linting error, it doesn't enforce the contract specified by the documentation.

Thoughts?

Reply:
If people implement it with an arrow function, the error will be gone on their side as well, no?

Reply:
It might be that my example of adding the `this.deleteData(id)` into the object creation of `createWorkersKVSessionStorage()` is too contrived.

This would mostly matter if the implementer is creating a class that implements the `SessionIdStorageStrategy` interface and passes that as the strategy to `createSessionStorage`. It appears that a class does not pick up on the difference one way or the other, arrow function or not.

I reverted the changes to `SessionIdStorageStrategy` and converted the `SessionStorage` functions to arrow functions as suggested.